test/date-time/date-time-utility.test.ts

```diff
@@ -0,0 +1,397 @@
+import * as Assert from "assert";
+import { DateTime } from "../../src/date-time";
+import { IANAZone, DateTime as LuxonDateTime } from "luxon";
+
+
+suite("DateTime Utility", () =>
+{
+    suite("Current zone", () =>
+    {
+        test(`Given current system zone from DateTime
+        when it's validated that it's a proper zone
+        then it should return true`,
+            () =>
+            {
+                Assert.ok(DateTime.validateTimeZone(DateTime.currentZone));
+            }
+        );
+
+        test(`Given current system zone from DateTime
+        when it's compared to luxon current system zone
+        then it should be same as Luxon local zone`,
+            () =>
+            {
+                Assert.strictEqual(LuxonDateTime.now().zoneName, DateTime.currentZone);
+            }
+        );
+    });
+
+    suite("Value of", () =>
+    {
+        const value = "2024-01-01 10:00";
+        const zone = "utc";
+        const dateTime = new DateTime({ value: value, zone: zone });
+        const luxonDateTime = LuxonDateTime.fromFormat(value, "yyyy-MM-dd HH:mm", { zone });
+
+        test(`Given a DateTime with value "2024-01-01 10:00" and zone utc
+        when it's compared to luxon dateTime representing the same 
+        then it's valueOf() should be same as Luxon DateTime`,
+            () =>
+            {
+                Assert.strictEqual(dateTime.valueOf(), luxonDateTime.valueOf());
+            }
+        );
+
+        test(`Given a DateTime with value "2024-01-01 10:00" and zone utc
+        when it's valueOf() is validated with luxon
+        then it should return true`,
+            () =>
+            {
+                Assert.ok(LuxonDateTime.fromMillis(dateTime.valueOf()).isValid);
+            }
+        );
+
+        test(`Given a DateTime with value "2024-01-01 10:00" and zone utc
```

Comment:
What are you testing here?

Reply:
valueOf( should return milliSeconds since epoch which should be timestamp *1000